distributed/scheduler.py

```diff
@@ -654,6 +657,79 @@ def _to_dict_no_nest(self, *, exclude: Container[str] = ()) -> dict[str, Any]:
             members=True,
         )
 
+    def add_to_processing(self, ts: TaskState) -> None:
+        """Assign a task to this worker for compute."""
+        assert self.scheduler_ref and (scheduler := self.scheduler_ref())
+        tg = ts.group
+        self.task_groups_count[tg.name] += 1
+        scheduler.task_groups_count_global[tg.name] += 1
+        self.processing.add(ts)
+        for dts in ts.dependencies:
+            if self not in dts.who_has:
+                self._inc_needs_replica(dts)
+
+    def remove_from_processing(self, ts: TaskState) -> None:
+        """Remove a task from a workers processing"""
+        if ts not in self.processing:
+            return
+        assert self.scheduler_ref and (scheduler := self.scheduler_ref())
+        self.task_groups_count[ts.group.name] -= 1
+        scheduler.task_groups_count_global[ts.group.name] -= 1
+        if not self.task_groups_count[ts.group.name]:
+            del self.task_groups_count[ts.group.name]
+        if not scheduler.task_groups_count_global[ts.group.name]:
+            del scheduler.task_groups_count_global[ts.group.name]
```

Comment:
What's the purpose of these `del`s on a `defaultdict`?

Reply:
I don't want to iterate over them if they are empty